src/core/networkreply.h

```diff
@@ -58,7 +58,7 @@ class NetworkReply : public QObject
 
 
     /**
-     * Get the QNetworkReply object once the CloudReply is finilized. Do not delete it manually.
+     * Get the QNetworkReply object. Do not delete it manually.
```

Comment:
```suggestion
     * Get the current `QNetworkReply` object. Note that it might get deleted even if the parent `NetworkReply` is not in case of redirect or internal retry. Do not delete it manually.
```

Reply:
Thanks Ivan :+1:

Reply:
@suricactus  do you think should we handle it like this: 

```
    if( !rawReply ) {
      hasError = true;
      errorMessageDetail = "";
      errorMessage = tr( "" );
    }
```

and never call `rawReply->readAll()`.

Reply:
@mohsenD98 what if we rename the getter to `currentRawReply()` and add a new signal `currentRawReplyChanged()`, so the caller would know at any time if the reply has changed?

If you receive `NetworkReply::downloadProgress` most probably `currentRawReply()` will return an active pointer, if it doesn't, I think a simple `if (!rawReply) return` should be enough?

Reply:
@nirvn what do you think about this ?

Reply:
@suricactus , @mohsenD98 , this sounds good to me.